skiko/src/awtMain/objectiveC/macos/DisplayLinkThrottler.mm

```diff
@@ -162,13 +162,20 @@ JNIEXPORT void JNICALL Java_org_jetbrains_skiko_redrawer_DisplayLinkThrottler_di
 }
 
 JNIEXPORT void JNICALL Java_org_jetbrains_skiko_redrawer_DisplayLinkThrottler_waitVSync(JNIEnv *env, jobject obj, jlong throttlerPtr, jlong windowPtr) {
-    DisplayLinkThrottler *throttler = (__bridge DisplayLinkThrottler *) (void *) throttlerPtr;
-    NSWindow *window = (__bridge NSWindow *) (void *) windowPtr;
+    @autoreleasepool {
+        DisplayLinkThrottler *throttler = (__bridge DisplayLinkThrottler *) (void *) throttlerPtr;
 
-    // CVDisplayLink is conditionally set up on each draw request to track window.screen change to match throttling
-    // with the refresh rate of the actual screen the window is shown on.
-    [throttler setupDisplayLinkForWindow:window];
-    [throttler waitVSync];
+        NSWindow *window = nil;
+
+        if (windowPtr != 0) {
+            window = (__bridge_transfer NSWindow *) (void *) windowPtr;
+        }
+
+        // CVDisplayLink is conditionally set up on each draw request to track window.screen change to match throttling
+        // with the refresh rate of the actual screen the window is shown on.
+        [throttler setupDisplayLinkForWindow:window];
```

Comment:
If it is zero, then we don't wait for vsync? In which cases this can happen?

Reply:
I looked at the code, and it is written in consideration that `window` always exists in this moment:

```
    override fun dispose() = synchronized(drawLock) {
        frameDispatcher.cancel()
        contextHandler.dispose()
        disposeDevice(device.ptr)
        adapter.dispose()
        displayLinkThrottler.dispose()
        _device = null
        super.dispose()
    }

    private fun performDraw() = synchronized(drawLock) {
        if (!isDisposed) {
            ...
            displayLinkThrottler.waitVSync(windowPtr = layer.windowHandle)
            ...
        }
    }
```

`AWT` should retain link to the window until we call `super.dispose()` (or after it). Not sure if it is true in reality.

Reply:
Yeah, I don't know the reason for marking this specific spot as `weak` and there is nobody to ask.